src/DGtal/shapes/DigitalShapesDecorator.h

```diff
@@ -164,6 +164,7 @@ namespace DGtal
       if( !isValid() )
       {
         trace.error() << "Operation unvalid. Maybe you don't set a ShapeA object." << std::endl;
+        ASSERT(( isValid() ));
```

Comment:
kittens are dying... :crying_cat_face: 

Please consider DGtal::FATAL_ERROR or FATAL_ERROR_MSG to have a non-static assert even in release mode